apps/heft/src/plugins/CopyStaticAssetsPlugin.ts

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { LegacyAdapters, FileSystem, Terminal } from '@rushstack/node-core-library';
+import * as glob from 'glob';
+import * as globEscape from 'glob-escape';
+import * as path from 'path';
+import * as chokidar from 'chokidar';
+
+import { Async } from '../utilities/Async';
+import { performance } from 'perf_hooks';
+import { IPluginPackage } from '../pluginFramework/IPluginPackage';
+import { Build, HeftCompilation } from '../pluginFramework/HeftCompilation';
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { ICompilePhase, ICopyStaticAssetsConfiguration } from '../cli/actions/BuildAction';
+import { PrefixProxyTerminalProvider } from '../utilities/PrefixProxyTerminalProvider';
+
+const PLUGIN_NAME: string = 'CopyStaticAssetsPlugin';
+
+export const copyStaticAssetsPlugin: IPluginPackage = {
+  displayName: PLUGIN_NAME,
+  apply: (heftCompilation: HeftCompilation, heftConfiguration: HeftConfiguration) => {
```

Comment:
My overall feedback is that this model might make **CopyStaticAssetsPlugin.ts** easy to write, but it is somewhat intimidating for a newcomer to read. (And it's one of the simplest tasks!)  The reason is lack of structure.

Proposal: We should try to isolate all the tapping (i.e. registering event handlers) in a setup function that avoids doing any actual computation. The reason is that tapping characterizes how this plugin interacts with everything else, which is very important information.

Also, I'd prefer for the event handlers to be top-level members with meaningful names, rather than disembodied chunks of code.  This helps you to see the big operations we're performing, and it encourages documentation.  For example we might feel more guilty about not commenting the `compile.hooks.run.tapPromise()` if it was a named function declaration, rather than pretending to be a function call.

Reply:
(I'm not very experienced with tappable or Webpack plugins though, so if these suggestions actually make things worse, I'll reconsider.)